lib/annotate/annotate_models.rb

```diff
@@ -159,13 +159,15 @@ def serialize_files(root_directory)
       ]
     end
 
-    def files_by_pattern(root_directory, pattern_type)
+    def files_by_pattern(root_directory, pattern_type, options)
       case pattern_type
       when 'test'       then test_files(root_directory)
       when 'fixture'    then fixture_files(root_directory)
       when 'scaffold'   then scaffold_files(root_directory)
       when 'factory'    then factory_files(root_directory)
       when 'serializer' then serialize_files(root_directory)
+      when 'additional_file_patterns'
+        [options[:additional_file_patterns] || []].flatten
```

Comment:
Can you elaborate what's going on here?

Reply:
This will return the config setting `additional_file_patterns`, which is a list of files and/or globs.  This allows the gem to infer models based on file structure or naming and annotate additional files.

Reply:
I should've been more specific - I'm curious about:
```ruby
[options[:additional_file_patterns] || []].flatten
```
Is there any special reason behind it? Couldn't it also be:
```ruby
options[:additional_file_patterns] || []
```